### ISSUE TITLE
Support for Dokka

### ISSUE THREAD

**Issue by saket** [vanniktech/gradle-maven-publish-plugin#21]
Hey Niklas, will it be possible to add support for including javadoc generated using Dokka? Javadoc and sources for Kotlin files aren't included otherwise.

**Comment by vanniktech** [comment]
Sure, want to take a stab at it?

**Comment by saket** [comment]
That's what I initially wanted to try, but I'm quite bad at build tools. Would you be okay if I send potentially buggy patch and help me understand how things work?

**Comment by digitalbuddha** [comment]
Not sure if its the proper way but here's how I did it in the past https://github.com/NYTimes/Store/search?utf8=%E2%9C%93&q=dokka&type=

**Comment by vanniktech** [comment]
Don't have too much time currently but yes let's try to figure out a solution.

**Comment by saket** [comment]
I'll try sending a patch this weekend.

**Comment by saket** [comment]
Hey @vanniktech, just wanted to let you know that I tried my hands on this twice but failed terribly. Being new to Groovy and Gradle plugins, I found myself lost in the syntax. Based on @hzsweers's implementation in [AutoDispose](https://github.com/uber/AutoDispose/blob/master/gradle/gradle-mvn-push.gradle), here's what I could come up with so far: 

https://github.com/saket/gradle-maven-publish-plugin/blob/master/src/main/groovy/com/vanniktech/maven/publish/MavenPublishPlugin.groovy#L74

**Comment by vanniktech** [comment]
Yup also believe this should be easier now. Will also try to have a look at this next week.

**Comment by saket** [comment]
@gabrielittner that looks much better

**Comment by hzsweers** [comment]
I'll take a crack at this

**Comment by Egorand** [comment]
Any progress on this?

On a side note, kinda frustrated by the amount of support this library is getting, thinking about switching back to custom Gradle files.

**Comment by Egorand** [comment]
Oh, found it, thanks. I'll review.
